source/common/common/utility.cc

```diff
@@ -33,6 +142,38 @@ std::string DateFormatter::fromTime(time_t time) const {
   return std::string(&buf[0]);
 }
 
+std::string
+DateFormatter::fromTimeAndPrepareSubsecondOffsets(time_t time,
+                                                  SubsecondOffsets& subsecond_offsets) const {
+  tm current_tm;
+  gmtime_r(&time, &current_tm);
+
+  std::array<char, 1024> buf;
+  std::string formatted;
+
+  int32_t previous = 0;
```

Comment:
Nit: prefer using signed types where possible, since they describe the allowed range.